### PR TITLE
Refresh UI notifications automatically on update

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

File: BTCPayServer/Controllers/NotificationsController.cs [hunk 1]
```diff
@@ -24,18 +41,63 @@ public class NotificationsController : Controller
...
+
+        [HttpGet]
+        public async Task<IActionResult> GetNotificationDropdownUI()
+        {
+            return ViewComponent("NotificationsDropdown");
+        }
+
+
+        [HttpGet]
+        public async Task<IActionResult> SubscribeUpdates()
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/NotificationsController.cs:69]: pass CancellationToken here

File: BTCPayServer/Controllers/NotificationsController.cs [hunk 2]
```diff
@@ -24,18 +41,63 @@ public class NotificationsController : Controller
...
+            {
+                subscription =  _eventAggregator.Subscribe<UserNotificationsUpdatedEvent>(async evt =>
+                {
+                    if (evt.UserId == userId)
+                    {
+                        await websocketHelper.Send("update");
+                    }
+                });
+               
+                while (!HttpContext.RequestAborted.IsCancellationRequested)
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/NotificationsController.cs:89]: use the passed CancellationToken here

File: BTCPayServer/Controllers/NotificationsController.cs [hunk 3]
```diff
@@ -24,18 +41,63 @@ public class NotificationsController : Controller
...
+                {
+                    await Task.Delay(2000);
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/NotificationsController.cs:91]: pass it here as well.

**Comment by Kukks** [comment]
done @NicolasDorier

**Pull request merged**